sql/telemetry_derived/devtools_events_amplitude_v1/view.sql

```diff
@@ -0,0 +1,449 @@
+  FROM
+    `moz-fx-data-shared-prod.telemetry.event`
+  CROSS JOIN
+    UNNEST(
+      [
+        STRUCT(
+          "content" AS event_process,
+          payload.events.content AS events
+        ),
+        ("dynamic", payload.events.dynamic),
+        ("extension", payload.events.extension),
+        ("gpu", payload.events.gpu),
+        ("parent", payload.events.parent)
+      ]
+    )
+  CROSS JOIN
+    UNNEST(events) AS e
+), main_events AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    'main' AS doc_type,
+    document_id,
+    client_id,
+    normalized_channel,
+    normalized_country_code AS country,
+    environment.settings.locale AS locale,
+    normalized_app_name AS app_name,
+    metadata.uri.app_version AS app_version,
+    normalized_os AS os,
+    normalized_os_version AS os_version,
+    environment.experiments AS experiments,
+    sample_id,
+    payload.info.session_id AS session_id,
+    SAFE.PARSE_TIMESTAMP(
+      "%FT%H:%M:%S.0%Ez",
+      payload.info.session_start_date
+    ) AS session_start_time,
+    payload.info.subsession_id AS subsession_id,
+    submission_timestamp AS `timestamp`,
+    `moz-fx-data-shared-prod.udf.deanonymize_event`(e).*,
+    event_process,
+    environment.build.architecture AS env_build_arch,
+    application.build_id AS application_build_id,
+    environment.settings.is_default_browser,
+    environment.system.is_wow64,
+    environment.system.memory_mb,
+    environment.profile.creation_date AS profile_creation_date,
+    environment.settings.attribution.source AS attribution_source,
+    metadata.geo.city
+  FROM
+    `moz-fx-data-shared-prod.telemetry.main`
+  CROSS JOIN
+    -- While there are more "events" fields under other process in the main ping schema,
+    -- events were moved out to the event ping before those other processes were added. This is
+    -- an exhaustive list of processes in which we'd expect to see events in main pings
+    UNNEST(
+      [
+        STRUCT(
+          "content" AS event_process,
+          payload.processes.content.events AS events
+        ),
+        ("dynamic", payload.processes.dynamic.events),
+        ("gpu", payload.processes.gpu.events),
+        ("parent", payload.processes.parent.events)
+      ]
+    )
+  CROSS JOIN
+    UNNEST(events) AS e
+), events AS (
+  SELECT
+    *
+  FROM
+    main_events
+  UNION ALL
+  SELECT
+    *
+  FROM
+    event_events
+), base_events AS (
+  SELECT
+    *,
+    timestamp AS submission_timestamp,
+    event_string_value AS event_value,
+    UNIX_MILLIS(session_start_time) AS created
+  FROM
+    events
+), all_events AS (
+SELECT
+    submission_timestamp,
+    client_id AS user_id,
+    (created + COALESCE(SAFE_CAST(`moz-fx-data-derived-datasets.udf.get_key`(event_map_values, 'session_id') AS INT64), 0)) AS session_id,
+    CASE
+        WHEN (event_category IN ('devtools.main') ) AND (event_method IN ('open') ) AND (event_object IN ('tools') ) THEN 'dt - open' 
```

Comment:
It seems like we could simplify the case statement pretty drastically, e.g.
```
   CASE
        WHEN (event_object IN ('tools') ) THEN CONCAT('dt - ',  event_method)
        ELSE CONCAT('dt_', event_object, ' - ',  event_method)
    END
```
as long as we add into the `WHERE` clause something like `event_category IN ('devtools.main')`

My understanding is that they do want every devtools event in amplitude, and this way we mostly won't need to worry about adding new events in the future (I can confirm whether this makes sense in the channel)

Reply:
As discussed in #devtools-amplitude, we need to ignore `edit_rule` and `tool_timer` events and otherwise everything else can go through

Reply:
I'll file a separate PR for this change.